### PR TITLE
COM-3560: Change to accept custom headers from API calls

### PR DESCRIPTION
**Pull request by raghavendra0520**
Added a new property customHeaders to apiContext to send in the custom headers. These headers will be added to existing headers during api calls.

### CODE REVIEW

File: Mozu.Api/ApiContext.cs [hunk 1]
```diff
@@ -212,7 +217,8 @@ public IApiContext CloneWith(Action<IApiContext> contextModification)
                 NoCacheUpdate = (!String.IsNullOrEmpty(this.NoCacheUpdate) ? (string)this.NoCacheUpdate.Clone() : null),
                 BypassCache = (!String.IsNullOrEmpty(this.BypassCache) ? (string)this.BypassCache.Clone() : null),
                 Pricelist = (!String.IsNullOrEmpty(this.Pricelist) ? (string)this.Pricelist.Clone() : null),
-                PreviewDate = (!String.IsNullOrEmpty(this.PreviewDate) ? (string)this.PreviewDate.Clone() : null)
+                PreviewDate = (!String.IsNullOrEmpty(this.PreviewDate) ? (string)this.PreviewDate.Clone() : null),
+				CustomHeaders= (this.CustomHeaders != null) ? (new Dictionary<string,string>(this.CustomHeaders)) : null
```
Comment (rocky0904) [inline Mozu.Api/ApiContext.cs:221]: what is this doing here? not cloning the custom headers?

File: Mozu.Api.Test/MsTestCases/CustomHeaderTest.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        [Description("Custom Headers test")]
+        public void Http_Request_Contains_Custom_Header()
+        {
+            var customHeaders = new Dictionary<string, string>()
+            {
+                {"x-vol-cluster","unified" }
+            };
+            var baseUrlAddress = Helpers.Environment.GetConfigValueByEnvironment(Environment, "BaseAuthAppUrl");
+            SetSdKparameters();
+            var ApiMsgHandler = ServiceClientMessageFactory.GetTestClientMessage(tenantId, customHeaders);
```
Comment (rocky0904) [inline Mozu.Api.Test/MsTestCases/CustomHeaderTest.cs:29]: This is testing constructor instantiation. Add another test to modify existing api context via clone.

File: Mozu.Api/ApiContext.cs (hunk 1 above)
Comment (rocky0904) [reply Mozu.Api/ApiContext.cs:221]: add a test to validate the clone is working.

**Review by rocky0904 — CHANGES_REQUESTED** [review]

File: Mozu.Api.Test/MsTestCases/CustomHeaderTest.cs (hunk 2 above)
Comment (raghavendra0520) [reply Mozu.Api.Test/MsTestCases/CustomHeaderTest.cs:29]: Added couple test cases to test apiContext cloning.

File: Mozu.Api/ApiContext.cs (hunk 1 above)
Comment (raghavendra0520) [reply Mozu.Api/ApiContext.cs:221]: Added tests to validate apiContext cloning.

**Review by rocky0904 — APPROVED** [review]

**Pull request merged**